components/Radio/src/Radio.cpp

```diff
@@ -11,58 +11,109 @@
 
 #include "Radio.h"
 
+#include "freertos/queue.h"
 #include <driver/uart.h>
 #include <esp_log.h>
+#include <string.h>
+
+
+
+#define FTDI_PORT UART_NUM_0
+#define BUF_SIZE (1024)
+#define RD_BUF_SIZE (BUF_SIZE)
 
 namespace radio {
 
 #if RAD_ENABLED == true
 
 static const char* TAG = "UART";
+static QueueHandle_t uart0_queue;
 
 Radio::Radio()
     : Task("UART", 4096, 5)
 {
 }
 
-void Radio::run(void* data)
+void Radio::init(callbackFunction fn) 
```

Comment:
Give `callbackFunction` a more meaningful name and follow naming guidelines, like `radio_rx_cb_t`.